src/main/java/org/jabref/gui/texparser/ParseTexResultViewModel.java

```diff
@@ -1,24 +1,42 @@
 package org.jabref.gui.texparser;
 
+import java.util.ArrayList;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+import javafx.beans.property.BooleanProperty;
 import javafx.beans.property.ReadOnlyListWrapper;
+import javafx.beans.property.SimpleBooleanProperty;
 import javafx.collections.FXCollections;
 import javafx.collections.ObservableList;
 
 import org.jabref.gui.AbstractViewModel;
+import org.jabref.gui.importer.ImportEntriesDialog;
+import org.jabref.gui.util.BackgroundTask;
+import org.jabref.logic.l10n.Localization;
+import org.jabref.model.database.BibDatabaseContext;
 import org.jabref.model.texparser.Citation;
-import org.jabref.model.texparser.TexParserResult;
+import org.jabref.model.texparser.TexBibEntriesResolverResult;
 
 public class ParseTexResultViewModel extends AbstractViewModel {
 
+    private final TexBibEntriesResolverResult resolverResult;
+    private final BibDatabaseContext databaseContext;
     private final ObservableList<ReferenceViewModel> referenceList;
     private final ObservableList<Citation> citationList;
+    private final BooleanProperty importButtonDisabled;
 
-    public ParseTexResultViewModel(TexParserResult texParserResult) {
+    public ParseTexResultViewModel(TexBibEntriesResolverResult resolverResult, BibDatabaseContext databaseContext) {
+        this.resolverResult = resolverResult;
+        this.databaseContext = databaseContext;
         this.referenceList = FXCollections.observableArrayList();
         this.citationList = FXCollections.observableArrayList();
 
-        texParserResult.getCitations().asMap().forEach((entry, citations) ->
-                referenceList.add(new ReferenceViewModel(entry, citations)));
+        Set<String> newEntryKeys = resolverResult.getNewEntries().stream().map(entry -> entry.getCiteKeyOptional().orElse("")).collect(Collectors.toSet());
+        resolverResult.getCitations().asMap().forEach((entry, citations) ->
```

Comment:
Please extract the for each loop to a separate old school for each loop, it's better to read